registry/storage/driver/storagedriver.go

```diff
@@ -88,8 +88,14 @@ type StorageDriver interface {
 	// from the given path, calling f on each file.
 	// If the returned error from the WalkFn is ErrSkipDir and fileInfo refers
 	// to a directory, the directory will not be entered and Walk
-	// will continue the traversal.  If fileInfo refers to a normal file, processing stops
+	// will continue the traversal.
+	// If the returned error from the WalkFn is ErrFilledBuffer, processing stops.
 	Walk(ctx context.Context, path string, f WalkFn) error
+
+	// WalkWithStartAfterHint traverses a filesystem defined within driver as in
+	// Walk. If startAfterHint is set, the walk may start with the first item lexographically
+	// after the hint, but it is not guaranteed and drivers may start the walk from the path.
+	WalkWithStartAfterHint(ctx context.Context, path string, startAfterHint string, f WalkFn) error
```

Comment:
I wonder if it'd be better to pass some functional opts to `Walk` i.e. `WithHints` or some such rather than introducing another `interface` func to the already bloated `StorageDriver` interface.

I do like the approach in this PR, though 👍

Reply:
Do you mean something like this?

```
// WalkHints provides hints to the walk function to help drivers return only required
// items for processing
type WalkHints struct {
	// If startAfterHint is set, the walk may start with the first item lexographically
	// after the hint, but it is not guaranteed and drivers may start the walk from the path.
	StartAfter string
}
```
and:
```
Walk(ctx context.Context, path string, hints *WalkHints, f WalkFn) error
```

Reply:
Oh, functional opts, so sort of :)

Will code it up.

Reply:
Done. Feels cleaner. Good shout.